[4.0] "Authentification" correction

particthistle opened the pull request:
Correction of language and hint positioning for two factor authentication field on login page.

Pull Request does not have a related issue - I identified issue while checking through translations, then found an issue where it appears in the site.

### Summary of Changes
- Correct "Authentification" to "Authentication" in COM_LOGIN_TWOFACTOR string (Authentification is valid, but not proper en_GB)
- On viewing where the string was being used, I then found the layout is incorrect, so have fixed that also.
- scss update to create new class for .form-control-hint to fix the layout.

### Testing Instructions
- In Joomla 4, enable one of the two factor authentication plugins
- Log out
- On the log in screen you'll now see the Secret Key box, with the spelling issue and the incorrect formatting of the hint that uses COM_LOGIN_TWOFACTOR string.
- Apply PR
- Log out again and check that the Log in screen now has the corrected en_GB spelling as well as better formatting per the AFTER result below.

### Actual result BEFORE applying this Pull Request
![Before-autheitification](https://user-images.githubusercontent.com/5515866/96845857-e25afe80-149c-11eb-977c-a618df0238e1.jpg)
- Spelling error still visible - COM_LOGIN_TWOFACTOR string will display as "For Two-Factor Authentification" for en_GB
- Hint for Secret Key is supposed to be positioned on the right, but due to double span the text-right class fails to do so. 

### Expected result AFTER applying this Pull Request
![image](https://user-images.githubusercontent.com/5515866/96849525-6f07bb80-14a1-11eb-8210-4123ffb63a76.png)
- COM_LOGIN_TWOFACTOR string will now show as "For Two-Factor Authentication"
- Hint for Secret Key will now be displayed above the field, to the right of the field.

### Documentation Changes Required
No. en_GB translation will be updated, but will most likely have been correctly translated into other languages previously. I just spotted the issue when translating en_US.

particthistle committed:
Authentification correction

Correction of language and hint positioning for two factor authentication field on login page.

particthistle committed:
Revert "Authentification correction"

This reverts commit aed9d1d6800464f1b340fd81d47850c6608505da.

particthistle committed:
Revert "Revert "Authentification correction""

This reverts commit 46d22e95ca3d018f76a83579003a732591dbe1b0.

brianteeman commented:
LGTM

particthistle commented:
I've checked I got the SCSS change correct this time by applying the PR package to my test site. Result was as expected in the AFTER result, so I look forward to seeing successful tests shortly.

richard67 commented:
@particthistle SCSS code style check fails in Drone, error log see here: https://ci.joomla.org/joomla/joomla-cms/36866/1/22

particthistle committed:
Update _login.scss

Corrected SCSS errors.

particthistle committed:
Update _login.scss

SCSS fix

richard67 commented:
@particthistle Now drone says in the npm clone step:

```
CSS file copied/minified: /********/src/build/media_source/system/css/fields/calendar.css
--
Error: Invalid CSS after "}": expected "}", was ""
        on line 251 of administrator/templates/atum/scss/blocks/_login.scss
        from line 43 of administrator/templates/atum/scss/template.scss
>> }
```

The changes on GitHub for that file show up as:

![2020-10-22_01](https://user-images.githubusercontent.com/7413183/96859936-693eb580-1462-11eb-8a0b-81c973037481.png)

Can it be that there is missing a closing curly bracket before the `.form-control-hint`?

Beside that, check code style. In scss we use 2 spaces for each level of indentation and no tabs.

If that helps, adjust your editor so it shows tabs and spaces, then you can see what is used.

particthistle committed:
Fix SCSS

richard67 commented on administrator/templates/atum/scss/blocks/_login.scss:
```diff
@@ -60,6 +60,13 @@
           bottom: -1.1rem;
           font-size: .75rem;
           text-align: right;
+        }
+		.form-control-hint {
+          position: absolute;
+          top: .3rem;
+		  right: 0;  
+          font-size: .75rem;
+          text-align: right;
```
```suggestion
        }

        .form-control-hint {
          position: absolute;
          top: .3rem;
          right: 0;  
          font-size: .75rem;
          text-align: right;
```
Fix code style.

particthistle committed:
Update _login.scss

bembelimen commented on administrator/templates/atum/scss/blocks/_login.scss:
```diff
@@ -61,6 +61,13 @@
+          position: absolute;
+          top: .3rem;
+          right: 0;
+          font-size: .75rem;
```
Probably you should use this variable to be in sync: https://github.com/twbs/bootstrap/blob/main/scss/forms/_form-text.scss#L7

bembelimen replied:
See: https://github.com/joomla/joomla-cms/pull/31191

particthistle replied:
#31191 was resolved by #31207. Changes made to _login.css in commit below follow variable structure from #31207 for all the font-size items in that file.

ceford commented:
I have tested this item :white_check_mark: successfully on 80e1daf4a70b7f2c869045e8413f0819e70421cf<br /><br />Tip: use a different browser to see Login form - no need to logout.
Needs npm ci
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31202">issues.joomla.org/tracker/joomla-cms/31202</a>.</sub>

particthistle committed:
Change SCSS font sizes to use variable as base

Following comment from @bembelimen
Applies changes made in #31207 to _login.scss.

particthistle committed:
Merge remote-tracking branch 'upstream/4.0-dev' into 4.0-dev

particthistle committed:
Fix spacing in scss

SharkyKZ commented on .gitignore:
```diff
@@ -107,3 +107,5 @@ RoboFile.ini
 # Media Manager
 /media/com_media/js/mediamanager.min.js.map
 /media/com_media/css/mediamanager.min.css.map
+administrator/templates/atum/scss/blocks/_notes/dwsync.xml
+administrator/templates/atum/scss/template.css
```
Looks like these were added by mistake.

```suggestion
```

particthistle replied:
Can anyone point me to info on how to go about removing files from a commit in GitHub?

I had to redo the initial commits several times due to it wanting to merge the entire fork :(

particthistle committed:
Removed accidental files.

Co-authored-by: SharkyKZ <sharkykz@gmail.com>

particthistle committed:
Merge pull request #1 from joomla/4.0-dev

4.0 dev

particthistle committed:
Update plg_content_fields.ini

particthistle committed:
Revert "Update plg_content_fields.ini"

This reverts commit 6596a491d572ce8a12604321529efd88d4e3c4f4.

richard67 commented:
@ceford Could you test again? Thanks in advance.

ceford commented:
I have tested this item :white_check_mark: successfully on 66dc2dcad7802b6968fce0319517f4085448d2a0<br /><br />Seems OK but I did not see the Web Authentication bar with either plugin.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31202">issues.joomla.org/tracker/joomla-cms/31202</a>.</sub>

ceford commented:
![screen shot 2020-11-01 at 19 03 07](https://issues.joomla.org/uploads/1/ef2614d6dae9a3f9a2bce98155d63962.png)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31202">issues.joomla.org/tracker/joomla-cms/31202</a>.</sub>

richard67 commented:
@ceford The web authentication button is only shown when accessing the site with https. When accessing with http it is not shown How did you test? https or http?.

ceford commented:
Using localhost - and access is via http.

richard67 commented:
> Using localhost - and access is via http.

@ceford If localhost or other host doesn't matter. But as I wrote it is desired behaviour that the web authentication button is only shown with https and not with http.

brianteeman commented:
More than desired it is intentional as webauthn will only work on https

ceford commented:
Thanks for the tips! I did not have https enabled on localhost. I have now done that and see the Web Authentication button. Next job for me: find out what it does!

gostn commented:
I have tested this item :white_check_mark: successfully on 66dc2dcad7802b6968fce0319517f4085448d2a0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31202">issues.joomla.org/tracker/joomla-cms/31202</a>.</sub>

zero-24 commented:
> Next job for me: find out what it does!

Making your site more secure :-D You can check here as a starting point: https://en.m.wikipedia.org/wiki/WebAuthn

richard67 commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31202">issues.joomla.org/tracker/joomla-cms/31202</a>.</sub>

chmst commented:
Thanks!
